src/main/java/com/bittercode/service/impl/BookServiceImpl.java

```diff
@@ -36,23 +37,20 @@ public class BookServiceImpl implements BookService {
             + BooksDBConstants.COLUMN_QUANTITY + "=? "
             + "  WHERE " + BooksDBConstants.COLUMN_BARCODE
             + "=?";
-
     @Override
     public Book getBookById(String bookId) throws StoreException {
-        Book book = null;
+        Book book = new Book();
```

Comment:
This definitely changes the semantics here. I would expect that the correct thing to do is to return null when the book isn't found, but it looks like you've changed this method so that it instead returns a nonnull, empty `Book`.